src/battle_util.c

```diff
@@ -3958,23 +3958,14 @@ bool32 ChangeTypeBasedOnTerrain(u32 battler)
     return TRUE;
 }
 
-// Supreme Overlord adds a damage boost for each fainted ally.
-// The first ally adds a x1.2 boost, and subsequent allies add an extra x0.1 boost each.
+// Supreme Overlord adds a x0.1 damage boost for each fainted ally.
 static uq4_12_t GetSupremeOverlordModifier(u32 battler)
 {
     u32 i;
-    struct Pokemon *party = GetBattlerParty(battler);
     uq4_12_t modifier = UQ_4_12(1.0);
-    bool32 appliedFirstBoost = FALSE;
 
-    for (i = 0; i < PARTY_SIZE; i++)
-    {
-        if (GetMonData(&party[i], MON_DATA_SPECIES) != SPECIES_NONE
-         && !GetMonData(&party[i], MON_DATA_IS_EGG)
-         && GetMonData(&party[i], MON_DATA_HP) == 0)
-            modifier += (!appliedFirstBoost) ? UQ_4_12(0.2) : UQ_4_12(0.1);
-        appliedFirstBoost = TRUE;
-    }
+    for (i = 0; i < gBattleStruct->supremeOverlordCounter[battler]; i++)
+        modifier += UQ_4_12(0.1);
```

Comment:
Would `modifier += UQ_4_12(0.1) * gBattleStruct->supremeOverlordCounter[battler]` not work?

Reply:
Yes, that's better. Will add it.